src/components/util.ts

```diff
@@ -71,9 +73,34 @@ const CAN_HAVE_CHILDREN = [
 
 export type ParentNodeTypes = typeof CAN_HAVE_CHILDREN[number]
 
-export const canNodeHaveChildren = (type: NodeType) => (CAN_HAVE_CHILDREN as readonly string[]).includes(type)
+export const canNodeHaveChildren = (type: NodeType) =>
+  (CAN_HAVE_CHILDREN as readonly string[]).includes(type)
 
 export const capitalize = (s: string) => {
   if (!s) return ""
   return s[0].toLocaleUpperCase() + s.slice(1)
 }
+
+/**
+ * Returns the same reference to `value` as long as the the given values are
+ * deep equal.
+ *
+ * This is, in general, not a great idea. See
+ * https://github.com/facebook/react/issues/14476#issuecomment-471199055
+ *
+ * This should work OK for mathbox options because the objects in question are
+ * either shallow, or are deep equal at a fairly shallow level.
+ * E.g., OrbitControls may not be a shall object, but if it's not, it will at
+ * least be deep equal between renders.
+ *
+ * But should work OK for mathbox options.
+ */
+export const useDeepCompareMemo = <T>(value: T, initial: T): T => {
+  const oldValue = useRef<T>(initial)
+  const memoOptions = useMemo(() => {
+    if (isEqual(value, oldValue.current)) return oldValue.current
```

Comment:
I'm not sure how lodash's `isEqual` works, but hopefully it is first checking reference equality, and then falling back to deep equality? That would let us keep performance in most cases, and only defer to deep equality when we've already potentially committed to an expensive re-render (and can reap the benefits of skipping it).

Reply:
> I'm not sure how lodash's isEqual works, but hopefully it is first checking reference equality, and then falling back to deep equality? 

I assume it does, but I don't know for sure. If it doesn't the issue is isolated to here finding a more efficient deep equal

But... lodash is so widely used, I feel it must.